tests/foreman/cli/test_errata.py

```diff
@@ -639,6 +643,186 @@ def test_positive_list_affected_chosts_by_erratum_restrict_flag(self):
+        assert self.virtual_machines[1].hostname not in result
+        # Step 2: Search for hosts that require RHSA errata
+        result = Host.list(
+            {
+                'search': f'errata_status = security_needed',
+                'organization-id': self.org['id'],
+                'per-page': 1000,
+            }
+        )
+        result = [item['name'] for item in result]
+        # Assert first VM's name is not in the list
+        assert self.virtual_machines[0].hostname not in result
+        # Assert second VM's name is in the list
+        assert self.virtual_machines[1].hostname in result
+        # Step 3: Search for hosts that have RHBA-2012:1030 applicable
+        result = Host.list(
+            {
+                'search': f'applicable_errata = {FAKE_5_ERRATA_ID}',
+                'organization-id': self.org['id'],
+                'per-page': 1000,
+            }
+        )
+        result = [item['name'] for item in result]
+        # Assert first VM's name is in the list
+        assert self.virtual_machines[0].hostname in result
+        # Assert second VM's name is not in the list
+        assert self.virtual_machines[1].hostname not in result
+        # Step 4: Search for hosts that have RHSA-2012:0055 applicable
+        result = Host.list(
+            {
+                'search': f'applicable_errata = {FAKE_2_ERRATA_ID}',
+                'organization-id': self.org['id'],
+                'per-page': 1000,
+            }
+        )
+        result = [item['name'] for item in result]
+        # Assert first VM's name is not in the list
+        assert self.virtual_machines[0].hostname not in result
+        # Assert second VM's name is in the list
+        assert self.virtual_machines[1].hostname in result
+        # Step 5: Search for hosts that have RPM for RHBA-2012:1030 applicable
+        result = Host.list(
+            {
+                'search': f'applicable_rpms = {FAKE_5_CUSTOM_PACKAGE}',
+                'organization-id': self.org['id'],
+                'per-page': 1000,
+            }
+        )
+        result = [item['name'] for item in result]
+        # Assert first VM's name is in the list
+        assert self.virtual_machines[0].hostname in result
+        # Assert second VM's name is not in the list
+        assert self.virtual_machines[1].hostname not in result
+        # Step 6: Search for hosts that have RPM for RHSA-2012:0055 applicable
+        result = Host.list(
+            {
+                'search': f'applicable_rpms = {FAKE_2_CUSTOM_PACKAGE}',
+                'organization-id': self.org['id'],
+                'per-page': 1000,
+            }
+        )
+        result = [item['name'] for item in result]
+        # Assert first VM's name is not in the list
+        assert self.virtual_machines[0].hostname not in result
+        # Assert second VM's name is in the list
+        assert self.virtual_machines[1].hostname in result
```

Comment:
Good to reduce the line of code using iteration over the package list or tuple. Personally I think I would avoid so many comments. It seems not much required as almost time you are doing same kind of assertions

Reply:
I have removed comments before asserts.